server/src/testFixtures/java/io/crate/testing/auth/TestJWTAuthenticationMethod.java

```diff
@@ -0,0 +1,144 @@
+/*
+ * Licensed to Crate.io GmbH ("Crate") under one or more contributor
+ * license agreements.  See the NOTICE file distributed with this work for
+ * additional information regarding copyright ownership.  Crate licenses
+ * this file to you under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.  You may
+ * obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.  See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ *
+ * However, if you have executed another commercial license agreement
+ * with Crate these terms will supersede the license and you may use the
+ * software solely pursuant to the terms of the relevant commercial agreement.
+ */
+
+package io.crate.testing.auth;
+
+import static io.crate.role.metadata.RolesHelper.getSecureHash;
+import static io.crate.role.metadata.RolesHelper.userOf;
+import static io.crate.testing.auth.RsaKeys.PUBLIC_KEY_256;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+import java.security.KeyFactory;
+import java.security.PublicKey;
+import java.security.spec.EncodedKeySpec;
+import java.security.spec.X509EncodedKeySpec;
+import java.time.Clock;
+import java.time.LocalDate;
+import java.time.LocalDateTime;
+import java.time.Month;
+import java.util.Base64;
+import java.util.HashSet;
+import java.util.Set;
+
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+import com.auth0.jwk.Jwk;
+import com.auth0.jwk.JwkProvider;
+import com.auth0.jwt.interfaces.DecodedJWT;
+
+import io.crate.auth.JWTAuthenticationMethod;
+import io.crate.role.JwtProperties;
+import io.crate.role.Role;
+import io.crate.role.Roles;
+
+/**
+ * Test implementation, allowing to supply own keys.
+ * <br />
+ *
+ * Public key and token, signed by its counterpart private key
+ * are generated via https://jwt.io/#debugger-io.
+ * Token contains dummy key id (kid) which is assigned to a public key.
+ * exp is NumericDate which is epoch seconds, see https://www.rfc-editor.org/rfc/rfc7519#section-4.1.4
+ * <br />
+ *
+ * Decoded token:
+ * Header:
+ * {
+ *   "alg": "RS256",
+ *   "typ": "JWT",
+ *   "kid": "1"
+ * }
+ * Payload:
+ * {
+ *   "iss": "https://console.cratedb-dev.cloud/api/v2/meta/jwk/",
+ *   "username": "cloud_user",
+ *   "exp": 1893456000
+ * }
+
+ */
+public class TestJWTAuthenticationMethod extends JWTAuthenticationMethod {
+
+    // A dummy value which would normally pass in all happy path scenario.
+    public static final LocalDateTime EXPIRES_AT = LocalDate.of(2030, Month.JANUARY, 1).atStartOfDay();
```

Comment:
Maybe make this relative to today instead of hardcoding expiration at 2030

Reply:
`exp` is included in static token which is used in many tests (encoded into JWT_TOKEN constant). 
Making `exp` dynamic would case signing payload each time  - I think doesn't worth it